flambe/metric/metric.py

```diff
@@ -12,6 +13,18 @@ class Metric(Component):
     examples and provide as output a processd list of the same size.
 
     """
+    def __init__(self, name: Optional[str] = None):
```

Comment:
giving a name to all metrics, while also allowing the input to trainer to be a dict seems redundant. I guess I understand now why you wanted to make it a list. Designwise I find this solution a bit akward. Why would a metric needs to know its own name you know? Any reason not to keep the status quo? No names in the Metric and Dict only as input? I don't really get the benefits that this change provides